frontend/src/main/components/RequestType/RequestTypeTable.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+
+import { useBackendMutation } from "main/utils/useBackend";
+import {
+  cellToAxiosParamsDelete,
+  onDeleteSuccess,
+} from "main/utils/RequestTypeUtils";
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function RequestTypeTable({ requesttype, currentUser }) {
+  const navigate = useNavigate();
+
+  const editCallback = (cell) => {
+    navigate(`/requesttypes/create/${cell.row.values.id}`);
+  };
+
+  // Stryker disable all : hard to test for query caching
+
+  // when delete success, invalidate the correct query key (depending on user role)
+  const apiEndpoint = hasRole(currentUser, "ROLE_PROFESSOR")
+    ? "/api/requesttypes/all"
+    : "/api/requesttypes/all";
```

Comment:
This code makes no sense at all.   Look at it... it's checking whether the user has ROLE_PROFESSOR and then setting the value of the variable to the *exact same value* in both cases?